### PR TITLE
Refactor labels injection and add tests.

### PR DESCRIPTION
**Pull request by droslean**
This patch only moves the label injection code to its own function and adds unit tests for it.

### CODE REVIEW

File: pkg/steps/template_test.go [hunk 1]
```diff
@@ -238,3 +242,114 @@ func TestOperateOnTemplatePods(t *testing.T) {
...
+			testID: "nil refs in jobspec, no injection expected",
+			jobSpec: &api.JobSpec{
+				JobSpec: downwardapi.JobSpec{
+					Refs: nil,
+				},
+			},
+			template: &templateapi.Template{
+				TypeMeta:   meta.TypeMeta{Kind: "Template", APIVersion: "template.openshift.io/v1"},
+				ObjectMeta: meta.ObjectMeta{Name: "test-template"},
+				Objects: []runtime.RawExtension{
```
Comment (stevekuznetsov) [inline pkg/steps/template_test.go]: This field is not germane to this test. we should remove it

Comment (droslean) [reply pkg/steps/template_test.go]: addressed

File: pkg/steps/template_test.go [hunk 2]
```diff
@@ -238,3 +242,90 @@ func TestOperateOnTemplatePods(t *testing.T) {
...
+							Raw:    []byte(runtime.EncodeOrDie(corev1Codec, pod)),
+							Object: pod.DeepCopyObject(),
+						}
+					}(),
+				},
+			},
+			expected: &templateapi.Template{
+				TypeMeta:   meta.TypeMeta{Kind: "Template", APIVersion: "template.openshift.io/v1"},
+				ObjectMeta: meta.ObjectMeta{Name: "test-template"},
+				Objects: []runtime.RawExtension{
```
Comment (stevekuznetsov) [inline pkg/steps/template_test.go]: I meant _any_ Object field . We just need to check that the template has ObjectLabels set, so all this is extra in every test case

Comment (droslean) [reply pkg/steps/template_test.go]: oops, yea. I removed the others as well.

**Pull request merged**